packages/datetime/src/dateInput.tsx

```diff
@@ -413,12 +421,23 @@ export class DateInput extends AbstractPureComponent2<IDateInputProps, IDateInpu
             // Support IE11 (#2924)
             relatedTarget = document.activeElement as HTMLElement;
         }
+        const eventTarget = e.target as HTMLElement;
         // Beware: this.popoverContentEl is sometimes null under Chrome
         if (
             relatedTarget == null ||
             (this.popoverContentEl != null && !this.popoverContentEl.contains(relatedTarget))
         ) {
-            this.handleClosePopover();
+            // Exclude the following blur operations that makes "body" the activeElement
+            // and would close the Popover unexpectedly
+            // - On disabled change months buttons
+            // - DayPicker day elements, their "blur" will be managed at its own onKeyDown
+            // event listener for improving UX
+            const isChangeMonthEvt = eventTarget.classList.contains("DayPicker-NavButton");
```

Comment:
```suggestion
            const isChangeMonthEvt = eventTarget.classList.contains(Classes.DATEPICKER_NAVBUTTON);
```

you would need to add this class to `common/classes.ts` in this package